crates/trie/sparse/src/trie.rs

```diff
@@ -64,14 +64,18 @@ impl TrieMasks {
 /// 2. Update tracking - changes to the trie structure can be tracked and selectively persisted
 /// 3. Incremental operations - nodes can be revealed as needed without loading the entire trie.
 ///    This is what gives rise to the notion of a "sparse" trie.
-#[derive(PartialEq, Eq, Default)]
+#[derive(PartialEq, Eq, Clone)]
 pub enum SparseTrie<P = DefaultBlindedProvider> {
     /// The trie is blind -- no nodes have been revealed
     ///
     /// This is the default state. In this state,
     /// the trie cannot be directly queried or modified until nodes are revealed.
-    #[default]
-    Blind,
+    Blind {
+        /// This is an optional field that can be used to store a previously allocated
+        /// trie. In these cases, the trie will still be treated as blind, but the allocated trie
+        /// will be reused if the trie becomes revealed.
+        allocated: Option<Box<RevealedSparseTrie<P>>>,
```

Comment:
it doesn't make sense to keep entire revealed sparse trie. let's only keep `nodes` and `values` hash maps and any other stuff that we are interested in. can be extracted into `SparseTrieNodes` or `SparseTrieState` struct or smth

Reply:
made `SparseTrieState` and put nodes, values, and mask hashmaps